# Graded assignment 1

Mark Klik & Misja Mikkers


# Packages

We need the package `tidyverse`

In [25]:
library(tidyverse) 

# Assignment 1

First, we generate some random data:

In [26]:
df1 <- as.data.frame(matrix(runif(20*2), ncol=20) )
head(df1)

V1        V2        V3        V4        V5        V6        V7       
1 0.4206504 0.4745417 0.7307129 0.2368384 0.4226366 0.5669675 0.8183943
2 0.8670497 0.9420153 0.3438697 0.9959931 0.8055860 0.1984244 0.5125790
  V8        V9        V10       V11       V12       V13       V14      
1 0.2586114 0.6788866 0.6622689 0.6267735 0.6291043 0.2887843 0.9803132
2 0.9244986 0.5477910 0.8317046 0.8568455 0.4787390 0.3810195 0.9999293
  V15       V16       V17       V18       V19       V20       
1 0.8614988 0.7058019 0.8285982 0.5279042 0.8738485 0.64830156
2 0.1175396 0.1079199 0.8395915 0.1246322 0.3818108 0.04209364

_df1_ is a wide dataframe. The goal of the assignment is to change _df1_ into a long dataframe _df2_. And show the first rows of the dataframe using `head(df2)`

The dataframe _df2_ should look like this:

| Key | Value      |
|-----|------------|
| V1  | 0.09413864 |
| V1  | 0.69782526 |
| V2  | 0.23460454 |
| V2  | 0.51520514 |
| V3  | 0.63181844 |
| V3  | 0.22807211 |

In [27]:
df2 <- df1 %>% gather(Key, Value, V1:V20)
  

head(df2)

Key Value    
1 V1  0.4206504
2 V1  0.8670497
3 V2  0.4745417
4 V2  0.9420153
5 V3  0.7307129
6 V3  0.3438697

## Assignment 2

We will generate some data:

In [28]:
df3 <- as.data.frame(read.table(header=TRUE, text='
      id        V_1       V_2                
          1  42.467573  -87.810047     
          2  42.049148  -88.273029
          3  39.110539  -90.324080
                       '))

head(df3)

id V_1      V_2      
1 1  42.46757 -87.81005
2 2  42.04915 -88.27303
3 3  39.11054 -90.32408

In [29]:
df4 <- as.data.frame(read.table(header=TRUE, text='
     V_3            V_4         id                 
      one         1h            1
      two         1h            1
      three       8h            2
      four        1h            4
'))

head(df4)

V_3   V_4 id
1 one   1h  1 
2 two   1h  1 
3 three 8h  2 
4 four  1h  4


You will have to merge these files, returning all rows and all columns from both _df3_ and _df5_. Where there are not matching values, return _NA_ for the missings. 

This should be the end result:

| id | V_1      | V_2       | V_3   | V_4 |
|----|----------|-----------|-------|-----|
| 1  | 42.46757 | -87.81005 | one   | 1h  |
| 1  | 42.46757 | -87.81005 | two   | 1h  |
| 2  | 42.04915 | -88.27303 | three | 8h  |
| 3  | 39.11054 | -90.32408 | NA    | NA  |
| 4  | NA       | NA        | four  | 1h  |



In [30]:
df5 <- merge(df3, df4, all.x=TRUE, all.y=TRUE)
df5

id V_1      V_2       V_3   V_4
1 1  42.46757 -87.81005 one   1h 
2 1  42.46757 -87.81005 two   1h 
3 2  42.04915 -88.27303 three 8h 
4 3  39.11054 -90.32408 NA    NA 
5 4        NA        NA four  1h

End of notebook